pennylane/ops/op_math/exp.py

```diff
@@ -280,3 +288,162 @@ def simplify(self):
         if isinstance(self.base, qml.ops.op_math.SProd):  # pylint: disable=no-member
             return Exp(self.base.base.simplify(), self.coeff * self.base.scalar)
         return Exp(self.base.simplify(), self.coeff)
+
+    def generator(self):
+        r"""Generator of an operator that is in single-parameter-form.
+
+        For example, for operator
+
+        .. math::
+
+            U(\phi) = e^{i\phi (0.5 Y + Z\otimes X)}
+
+        we get the generator
+
+        >>> U.generator()
+          (0.5) [Y0]
+        + (1.0) [Z0 X1]
+
+        """
+        if np.real(self.coeff) != 0 or not self.base.is_hermitian:
+            raise GeneratorUndefinedError(
+                f"Exponential with coefficient {self.coeff} and base operator {self.base} does not have a generator."
+            )
+        return self.base
+
+    @property
+    def parameter_frequencies(self):
+        r"""Returns the frequencies for each operator parameter with respect
+        to an expectation value of the form
+        :math:`\langle \psi | U(\mathbf{p})^\dagger \hat{O} U(\mathbf{p})|\psi\rangle`.
+
+        These frequencies encode the behaviour of the operator :math:`U(\mathbf{p})`
```

Comment:
missing `\hat{}` on U

Reply:
This is identical to the method on `Operation`, and it has now been changed so it just inherits it from `Operation`, so this is gone from `Exp` now.